### PR TITLE
Add YAML configuration for automated MSAL release

### PR DESCRIPTION
**Pull request by ameyapat**
## Proposed changes

Added yml file that configures pipeline to do the following : 

- Build universal framework
- Send it to code signing service to sign
- Extract version from CHANGELOG and update files containing version number
- Update Package.swift with released package location & checksum
- Release to Github with release notes and assets

## Type of change

- [x] Feature work
- [ ] Bug fix
- [x] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [x] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [ ] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -97,6 +97,17 @@ You can use [Carthage](https://github.com/Carthage/Carthage) to install `MSAL` b
 ```
 github "AzureAD/microsoft-authentication-library-for-objc" "master"
 ```
+### Using Swift Packages
+
```
Comment (kaisong1990) [inline README.md:101]: Maybe also add a link to Apple Doc

**Review by kaisong1990 — APPROVED** [review]
LGTM with a minor suggestion

File: azure_pipelines/spm-framework.yml [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+      Pattern: 'MSAL.zip'
+      signConfigType: 'inlineSignParams'
+      SessionTimeout: '60'
+      MaxConcurrency: '50'
+      MaxRetryAttempts: '5'
+      UseMinimatch: false
+      inlineOperation: |
+        [
+          {
+            "keyCode": "CP-233039-Apple",
```
Comment (oldalton) [inline azure_pipelines/spm-framework.yml:115]: Is this a secret?

File: azure_pipelines/spm-framework.yml [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+          platforms: [
+                .macOS(.v10_12),.iOS(.v11)
+          ],
+          products: [
+              .library(
+                  name: "MSAL",
+                  targets: ["MSAL"]),
+          ],
+          targets: [
+              .binaryTarget(name: "MSAL", url: "https://github.com/$(repositoryName)/releases/download/releaseTag1.2.3/MSAL.zip", checksum: "abcdefabcdef1234567890abcdef1234567890abcdef1234567890abcdef1234")
```
Comment (oldalton) [inline azure_pipelines/spm-framework.yml:267]: Does checksum need to be dynamic here?

File: MSAL/src/MSAL_Internal.h [hunk 4]
```diff
@@ -27,7 +27,7 @@
 
 #define MSAL_VER_HIGH       1
 #define MSAL_VER_LOW        1
-#define MSAL_VER_PATCH      13
+#define MSAL_VER_PATCH      14-SPMTest
```
Comment (oldalton) [inline MSAL/src/MSAL_Internal.h:30]: I assume this one is for testing purposes

File: azure_pipelines/spm-framework.yml [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+  - name: 'releaseVersion'
+    value: null
+  - name: 'commonCoreReleaseVersion'
+    value: null
+  - name: 'frameworkChecksum'
+    value: null
+  - name: 'repositoryName' # Name of the repository
+    value: 'AzureAD/microsoft-authentication-library-for-objc'
+  - name: 'repositoryBranch' # Name of the branch where version number will be updated for files that contain version number.
+    value: 'ameyapat/spm-framework-build'
```
Comment (oldalton) [inline azure_pipelines/spm-framework.yml:11]: I assume this is for testing purposes

**Review by oldalton — APPROVED** [review]

File: azure_pipelines/spm-framework.yml (hunk 2 above)
Comment (ameyapat) [reply azure_pipelines/spm-framework.yml:115]: Not a secret, it is just a code to specify what type of code signing needs to be performed by ESRP

File: azure_pipelines/spm-framework.yml (hunk 3 above)
Comment (ameyapat) [reply azure_pipelines/spm-framework.yml:269]: This is just a garbage checksum value I've added initially. This task will add Package.swift to the repo if it does not exist with this garbage value and in the next step the value will be replaced with the actual checksum of the signed & zipped framework.

File: MSAL/src/MSAL_Internal.h (hunk 4 above)
Comment (ameyapat) [reply MSAL/src/MSAL_Internal.h:30]: Yes, just for testing. Will clean it up once all testing is done.

File: azure_pipelines/spm-framework.yml (hunk 5 above)
Comment (ameyapat) [reply azure_pipelines/spm-framework.yml:11]: Yes, just for testing. Will replace with 'master' once all testing is done.

File: README.md (hunk 1 above)
Comment (ameyapat) [reply README.md:101]: Added link

**Pull request merged**